src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -140,6 +140,13 @@ public boolean match(final Unit unit) {
       }
     }
   };
+  public static final Match<Unit> UnitIsDefenselessTransport = new Match<Unit>() {
+    @Override
+    public boolean match(final Unit unit) {
+      final UnitAttachment ua = UnitAttachment.get(unit.getType());
+      return ua.getTransportCapacity() != -1 && ua.getIsSea() && ua.getRawDefense() == 0;
```

Comment:
This won't work in all cases. You can have transports that have support attachments providing defense bonus to each other which would make a stack of transports have more than 0 defense.